File: elixir/prime_factors/lib/pf/prime_factors.ex
defmodule PF.PrimeFactors do

  def prime_factors(num, candidate \\ 2) do
    cond do
      num < candidate -> []
      rem(num, candidate) == 0 ->
        [candidate] ++ prime_factors(div(num, candidate), candidate)
      true -> prime_factors(num, candidate + 1)
    end
  end

end

File: go/src/prime_factors/prime_factors.go
package prime_factors

func Factors(num int) []int {
  primes := []int{}
  for candidate := 2; candidate <= num; candidate++ {
    for ; num % candidate == 0; num /= candidate {
      primes = append(primes, candidate)
    }
  }
  return primes
}
